test/datagenerator/1800medicare.go

```diff
@@ -0,0 +1,161 @@
+					_, err = outf.WriteString(fmt.Sprintf("%-"+fmt.Sprint(f.l)+"s", f.v))
+					if err != nil {
+						panic(err)
+					}
+				}
+				_, err = outf.WriteString("\n")
+				if err != nil {
+					panic(err)
+				}
+			}
+		}
+
+		err = inf.Close()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	err = outf.Close()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func profile(hicn string, acoID string) record {
+	dobMin, _ := time.Parse("2006-01-02", "1900-01-01")
+	dobMax := time.Now().Add(-65 * 365 * 24 * time.Hour)
+
+	p := record{
+		hicn:       field{l: 11, v: hicn},                     // HICN
+		blk:        field{l: 10},                              // Beneficiary link key
+		firstName:  field{l: 30, v: randWord(1, 30)},          // Beneficiary first name
+		middleName: field{l: 30},                              // Beneficiary middle name
+		lastName:   field{l: 40, v: randWord(1, 30)},          // Beneficiary last name
+		dob:        field{l: 8, v: ccyymmdd(dobMin, dobMax)},  // Beneficiary date of birth
+		addr1:      field{l: 55, v: addr1()},                  // Beneficiary address line 1
+		addr2:      field{l: 55},                              // Beneficiary address line 2
+		addr3:      field{l: 55},                              // Beneficiary address line 3
+		city:       field{l: 40, v: randWord(1, 40)},          // Beneficiary city
+		state:      field{l: 2, v: "ST"},                      // Beneficiary state
+		zip5:       field{l: 5, v: "00000"},                   // Beneficiary first five digits of ZIP code
+		zip4:       field{l: 4},                               // Beneficiary last four digits of ZIP code
+		gender:     field{l: 1, v: oneOfStr("M", "F", "U")},   // Beneficiary gender
+		acoID:      field{l: 5, v: acoID},                     // ACO identifier
+		acoName:    field{l: 70, v: randWord(1, 66) + " ACO"}, // ACO legal name
+	}
+
+	return p
+}
+
+func records(profile record) []record {
+	// Create 0-5 suppression records for this profile
+	ct := rand.Intn(6)
+	encDtMin, encDtMax := time.Now().Add(-7*24*time.Hour), time.Now()
+	effDtMin, effDtMax := time.Now().Add(-7*24*time.Hour), time.Now().Add(3*24*time.Hour)
+	records := []record{}
+
+	for i := 0; i < ct; i++ {
+		r := profile
+		r.encDate = field{l: 8, v: ccyymmdd(encDtMin, encDtMax)} // Encounter date
+		r.effDate = field{l: 8, v: ccyymmdd(effDtMin, effDtMax)} // Beneficiary data sharing effective date
+		r.srcCode = field{l: 5, v: oneOfStr("1800", "")}         // Beneficiary data sharing source code
+		r.mechCode = field{l: 1, v: oneOfStr("T", "")}           // Beneficiary option data sharing decision mechanism code
+		r.prefInd = field{l: 1, v: oneOfStr("Y", "N", "")}       // Beneficiary data sharing preference indicator
+		r.saEffDate = field{l: 8}                                // Beneficiary substance abuse data sharing effective date
```

Comment:
same thought as the above comment, let's pass the date and a blank value for the r.saEffDate